src/validation.cpp

```diff
@@ -2958,10 +3155,18 @@ void CChainState::ReceivedBlockTransactions(const CBlock& block, CBlockIndex* pi
             CBlockIndex *pindex = queue.front();
             queue.pop_front();
             pindex->nChainTx = (pindex->pprev ? pindex->pprev->nChainTx : 0) + pindex->nTx;
+            // Make sure to remove from sets which are indexed by nSequenceId first...
+            bool was_header_candidate = setBlockIndexHeaderCandidates.erase(pindex);
```

Comment:
`setBlockIndexHeaderCandidates.erase(pindex) > 0` to improve readability and avoid implicit conversion.